acmw-db-app/components/AttendanceForm.jsx

```diff
@@ -53,11 +49,124 @@ const AttendanceForm = (props) => {
                 <span>
-                    <SubmitButton label="Submit" handleChange={handleSubmit}/>
+                    <SubmitButton 
+                        label="Submit" 
+                        eventCode={eventCode}
+                        firstName={firstName}
+                        lastName={lastName}
+                        onSubmit={setStatus}
+                        status={status}
+                    />
                 </span>
             </div>
         </div>
     )
 }
 
-export default AttendanceForm
\ No newline at end of file
+export default AttendanceForm
+
+const TextInput = (props) => {
+
+    const [id, setId] = useState(props.id);
+    const [active, setActive] = useState(false);
+    const [value, setValue] = useState("");
+    const [label, setLabel] = useState(props.label);
+    const [predicted, setPredicted] = useState(props.predicted);
+    const [type, setType] = useState(props.type);
+  
+    const changeValue = (event) => {
+      setValue(event.target.value);
+      props.onChange(event.target.value);
+    }
+  
+    const handleKeyPress = (event) => {
+      if (predicted && predicted.startsWith(value) && event.key === "Enter") {
+        setValue(predicted);
+        props.onChange(predicted);
+      }
+    }
+
+/* APPEARANCE LOGIC
+     if not active and value (user input) is empty
+      then display placeholder white and background lighter purpur
+
+     if not active and value is non-empty
+      then display value white and background purpur
+
+     if active (it is focused bc user clicked/tabbed to it)
+      then display label bright purpur; display value dark gray; hide placeholder
+  */
+ return (
+    <div className={`${styles.field} ${active ? styles.active : ( value && styles.filled )}`}>
+      { type === "text" ?
+        ( active &&
+          predicted &&
+          predicted.startsWith(value) ?
+          <p className={styles.predicted}>{predicted}</p> :
+          ( !value && <p className={styles.placeholder}>{label}</p> ) ) :
+          ( type === "Last Name" &&
+            !active &&
+            !value &&
+            <p className={styles.placeholder}>{label}</p> )}
+      <input
+        id={id}
+        type={type}
+        value={value}
+        onChange={event => changeValue(event)}
+        onKeyPress={event => handleKeyPress(event)}
+        onFocus={() => setActive(true)}
+        onBlur={() => setActive(false)}
+      />
+      <label htmlFor={id}>
+        {label}
+      </label>
+    </div>
+  );
+}
+
+const SubmitButton = (props) => {
+
+  const [message, setMessage] = useState("");
+
+    // If event code is valid, add a database entry, and display "Submitted successfully" message
+    // Else if the event code is not valid, display "Invalid event code" message
+
+    const validateSubmitButton = async () => {
+    const regex = new RegExp(/^[a-z ,.'-]+$/i); //tests for first name
+    const regexLast = new RegExp(/^[a-z ,.'-]+\.[1-9]+$/i); //tests for last name.#
```

Comment:
Need to include numbers with 0s. You could do `[1-9]([0-9]+)?` to have a number with no leading 0s